src/upcoming-components/TextareaField/TextareaField.tsx

```diff
@@ -1,215 +0,0 @@
-import FieldNote from '../../components/FieldNote';
-import Icon, { IconName } from '../../components/Icon';
-import Label from '../../components/Label';
-import Textarea from '../Textarea';
-
-export interface Props {
-  /**
-   * Aria-describedby id string
-   */
-  'aria-describedby'?: string;
-  /**
-   * CSS class names that can be appended to the component.
-   */
-  className?: string;
-  /**
-   * Disables the field and prevents editing the contents
-   */
-  disabled?: boolean;
-  /**
-   * FieldNote
-   * Used as helper text or error message
-   */
-  fieldNote?: string | ReactNode;
-  /**
-   * Toggles the visibility of the label. If hidden, the label text will still be accessible to assistive technologies
-   */
-  hideLabel?: boolean;
-  /**
-   * Screen reader text used for the button alongside the input field (i.e. show/hide password button)
-   */
-  fieldButtonAriaLabel?: string;
-  /**
-   * Text used for the button alongside the input field (i.e. show/hide password button)
-   */
-  fieldButtonText?: string;
-  /**
-   * Function passed down from higher level component to trigger on click function of text field button
-   */
-  fieldButtonOnClick?: MouseEventHandler;
-  /**
-   * Name of SVG icon (i.e. chevron-down, minus, warning)
-   */
-  iconName?: IconName;
-  /**
-   * HTML id for the component
-   */
-  id?: string;
-  /**
-   * Gives a hint as to the type of data needed for textarea
-   */
-  inputMode?:
-    | 'text'
-    | 'email'
-    | 'url'
-    | 'search'
-    | 'tel'
-    | 'none'
-    | 'numeric'
-    | 'decimal';
-  /**
-   * Error state of the form field
-   */
-  isError?: boolean;
-  /**
-   * HTML label text
-   */
-  label?: string;
-  /**
-   * Max number of characters for the text input
-   */
-  maxLength?: number;
-  /**
-   * HTML name attribute for the input
-   */
-  name?: string;
-  /**
-   * Function that runs on change of the input
-   */
-  onChange?: ChangeEventHandler;
-  /**
-   * String for the optional label. By default it is '(optional)'
-   */
-  optionalLabel?: string;
-  /**
-   * Placeholder attribute for input. Note: placeholder should be used sparingly
-   */
-  placeholder?: string;
-  /**
-   * Toggles the form control's interactivity. When `readOnly` is set to `true`, the form control is not interactive
-   */
-  readOnly?: boolean;
-  /**
-   * Indicates that field is required for form to be successfully submitted
-   */
-  required?: boolean;
-  /**
-   * String for the required label to add additional information if needed.
-   */
-  requiredLabel?: string;
-  /**
-   *  Specifies the visible number of lines in a text area
-   */
-  rows?: number;
-  /**
-   * Title attribute on input
-   */
-  title?: string;
-  /**
-   * Value passed down from higher levels for initial state
-   */
-  value?: string;
-  /**
-   * Default value passed down from higher levels for initial state
-   */
-  defaultValue?: string;
-}
-
-export const TextareaField = ({
```

Comment:
Think when it's time to build the `Textarea` component, the `TextareaField` may come naturally and could consider using our `Fieldset` component there. Don't think there's much value in these props that is not already come across